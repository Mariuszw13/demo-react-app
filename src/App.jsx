import React from 'react';
import './App.css';
import {
    BrowserRouter as Router,
    Route,
    Switch,
    Redirect
} from "react-router-dom";
import MainPage from "./containers/MainPage";
import LoginPage from "./containers/LoginPage";
import {CookiesProvider} from 'react-cookie';
import PrivateRoute from "./components/auth/PrivateRoute";
import {withLayout} from "./components/Layout";
import 'bootstrap/dist/css/bootstrap.min.css';

const App = () => (
    <CookiesProvider>
        <Router>
            <Switch>
                <Route path="/login" exact component={LoginPage}/>
                <PrivateRoute exact path="/" component={withLayout(MainPage)}/>
                <Redirect from="*" to="/"/>
            </Switch>
        </Router>
    </CookiesProvider>

)

export default App;