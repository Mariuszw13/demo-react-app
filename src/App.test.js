import React from "react";
import { render } from "@testing-library/react";
import App from "./App";

describe("<App />", () => {
    const { container } = render(<App />);

    it("should render properly and match snapshot", () => {
        expect(container.firstChild).toMatchSnapshot();
    });
});
