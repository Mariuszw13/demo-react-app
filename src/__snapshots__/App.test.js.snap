// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should render properly and match snapshot 1`] = `
<form
  class="sc-AxgMl iEzbDl"
>
  <div
    class="form-group row"
  >
    <label
      class="form-label col-form-label col-sm-2"
    >
      Email
    </label>
    <div
      class="col-sm-10"
    >
      <input
        class="form-control"
        placeholder="email"
        type="email"
        value=""
      />
    </div>
  </div>
  <div
    class="form-group row"
  >
    <label
      class="form-label col-form-label col-sm-2"
      for="formPlaintextPassword"
    >
      Password
    </label>
    <div
      class="col-sm-10"
    >
      <input
        class="form-control"
        id="formPlaintextPassword"
        placeholder="Password"
        type="password"
        value=""
      />
    </div>
  </div>
  <button
    class="sc-AxhUy icMVrQ btn btn-primary"
    type="button"
  >
    login
  </button>
</form>
`;
